Flaky spark-3.3-scala-2.13-java-8 tests due to `No space left on device`

### Apache Iceberg version

main (development)

### Query engine

Spark

### Please describe the bug 🐞

```
  ./gradlew -DsparkVersions=3.3 -DscalaVersion=2.13 -DhiveVersions= -DflinkVersions= :iceberg-spark:iceberg-spark-3.3_2.13:check :iceberg-spark:iceberg-spark-extensions-3.3_2.13:check :iceberg-spark:iceberg-spark-runtime-3.3_2.13:check -Pquick=true -x javadoc
  shell: /usr/bin/bash -e {0}
  env:
    SPARK_LOCAL_IP: localhost
    JAVA_HOME: /opt/hostedtoolcache/Java_Zulu_jdk/8.0.402-6/x64
    JAVA_HOME_8_X64: /opt/hostedtoolcache/Java_Zulu_jdk/8.0.402-6/x64
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100 62076  100 62076    0     0  1376k      0 --:--:-- --:--:-- --:--:-- 1409k
...
> Task :iceberg-spark:iceberg-spark-extensions-3.3_2.13:test
org.apache.iceberg.spark.extensions.TestCopyOnWriteMerge > testMergeWithConcurrentTableRefresh[catalogName = testhive, implementation = org.apache.iceberg.spark.SparkCatalog, config = {type=hive, default-namespace=default}, format = parquet, vectorized = true, distributionMode = none, branch = test] FAILED
    java.lang.AssertionError: 
    Expecting actual throwable to be an instance of:
      java.lang.IllegalStateException
    but was:
      org.apache.spark.SparkException: Writing job aborted
    	at org.apache.spark.sql.errors.QueryExecutionErrors$.writingJobAbortedError(QueryExecutionErrors.scala:767)
    	at org.apache.spark.sql.execution.datasources.v2.V2TableWriteExec.writeWithV2(WriteToDataSourceV2Exec.scala:409)
    	at org.apache.spark.sql.execution.datasources.v2.V2TableWriteExec.writeWithV2$(WriteToDataSourceV2Exec.scala:353)
    	...(41 remaining lines not displayed - this can be changed with Assertions.setMaxStackTraceElementsDisplayed)
        at org.apache.iceberg.spark.extensions.TestCopyOnWriteMerge.testMergeWithConcurrentTableRefresh(TestCopyOnWriteMerge.java:148)
Error: No space left on device : '/home/runner/runners/2.314.1/_diag/pages/197d7517-f10e-4508-9021-857e76a59e70_80d440e4-f54b-5560-0192-53fee83660bc_1.log'
```

--- a/spark/v3.3/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestCopyOnWriteMerge.java
+++ b/spark/v3.3/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestCopyOnWriteMerge.java
@@ -48,6 +48,7 @@ import org.apache.spark.sql.Encoders;
 import org.apache.spark.sql.internal.SQLConf;
 import org.junit.Assert;
 import org.junit.Assume;
+import org.junit.Ignore;
 import org.junit.Test;
 
 public class TestCopyOnWriteMerge extends TestMerge {
@@ -70,6 +71,7 @@ public class TestCopyOnWriteMerge extends TestMerge {
   }
 
   @Test
+  @Ignore // Ignored due to https://github.com/apache/iceberg/issues/10040
   public synchronized void testMergeWithConcurrentTableRefresh() throws Exception {
     // this test can only be run with Hive tables as it requires a reliable lock
     // also, the table cache must be enabled so that the same table instance can be reused

